### PR TITLE
feat: add sign in form validation

### PR DESCRIPTION
**Pull request by jiejasonliu**
## sign-in improvements
- add form validation to fields and login request
- `continue as guest` now signs in as `John Doe` for now
- improved button disabling logic (submit and continue as guest action now mutually exclusive)

## registration panel improvements
- add fade animation to registration panel
- autocomplete should only occur in the sign-in page now
- fix css selector specificity for disabled buttons

---

https://user-images.githubusercontent.com/29434693/193754749-b8113355-d31f-4519-b76f-b32a6ebd2d52.mp4

### CODE REVIEW

File: src/components/registration-panel/registration-panel.scss [hunk 1]
```diff
@@ -39,6 +39,20 @@
       margin-bottom: 50px;
     }
 
+    // all registration buttons
+    .registration-button {
+      border-color: $white;
+      color: $white;
+      padding: 10px;
+
+      &:not(.disabled) {
```
Comment (derricklee43) [inline src/components/registration-panel/registration-panel.scss:48]: didn't know `:not()` was a thing. That is pretty dope

File: src/pages/sign-in-page/sign-in-page.tsx [hunk 2]
```diff
@@ -2,42 +2,94 @@ import React, { useState } from 'react';
...
+
+    // simple client side verifications
+    {
+      const newFormError: SignInFormError = {};
+      console.log(email.length, password.length);
+      if (email.length === 0) {
+        newFormError.email ??= 'Email should not be empty.';
+      }
+      if (!email.includes('@')) {
+        newFormError.email ??= `This doesn't appear to be a valid email.`;
```
Comment (derricklee43) [inline src/pages/sign-in-page/sign-in-page.tsx:81]: Didn't know `??=` was a think either, lol

**Review by derricklee43 — APPROVED** [review]
Looks dope! LGMT!

File: src/components/registration-panel/registration-panel.scss (hunk 1 above)
Comment (jiejasonliu) [reply src/components/registration-panel/registration-panel.scss:48]: yesss, there's a bunch of niche pseudo selectors but this is one of the more underrated ones

File: src/pages/sign-in-page/sign-in-page.tsx (hunk 2 above)
Comment (jiejasonliu) [reply src/pages/sign-in-page/sign-in-page.tsx:81]: yeah!! it saves a lot of headache

**Pull request merged**